vdj_ann/src/annotate.rs

```diff
@@ -836,15 +894,16 @@ pub fn annotate_seq_core(
     while i < semi.len() {
         let mut j = i + 1;
         while j < semi.len() {
-            if semi[j].0 != semi[i].0 || semi[j].1 != semi[i].1 {
+            if semi[j].ref_id != semi[i].ref_id || semi[j].offset != semi[i].offset {
                 break;
             }
             j += 1;
         }
         for k1 in i..j {
             for k2 in i..j {
-                if semi[k1].1 + semi[k1].2 + semi[k1].3 == semi[k2].1 + semi[k2].2 + semi[k2].3
-                    && semi[k1].3 > semi[k2].3
+                if semi[k1].offset + semi[k1].tig_start + semi[k1].len
+                    == semi[k2].offset + semi[k2].tig_start + semi[k2].len
```

Comment:
This already makes the code a lot more readable. If we also abstract away these operations as functions, it would make it even better. :)

Reply:
This spot here in particular drives me crazy because the code just above here has already ensured that `semi[k1].offset == semi[k2].offset`.

See also two PRs later where I collapse the three different structs in this PR into a single type.